docs/source/links/state.md

```diff
@@ -0,0 +1,609 @@
+entry in the cache as long as you have its cache key. In contrast,
+`cache.readQuery` only lets you read from the root of your cache.
+
+Let's go back to our previous todo list example and see how `cache.readFragment`
+can help us toggle one of our todos as completed.
+
+```js
+const todos = {
+  Mutation: {
+    toggleTodo: (_, variables, { cache }) => {
+      const id = `TodoItem:${variables.id}`;
+      const fragment = gql`
+        fragment completeTodo on TodoItem {
+          completed
+        }
+      `;
+      const todo = cache.readFragment({ fragment, id });
+      const data = { ...todo, completed: !todo.completed };
+
+      // you can also do cache.writeData({ data, id }) here if you prefer
+      cache.writeFragment({ fragment, id, data });
+      return data;
+    },
+  },
+};
+```
+
+In order to toggle our todo, we need the todo & its status from the cache, which
+is why we call `cache.readFragment` & pass in a fragment to retrieve it. The
+`id` we're passing into `cache.readFragment` refers to its cache key. If you're
+using the `InMemoryCache` and not overriding the `dataObjectFromId` config
+property, your cache key should be `__typename:id`.
+
+To write the data to the cache, you can use either `cache.writeFragment` or
+`cache.writeData`. The only difference between the two is that
+`cache.writeFragment` requires that you pass in a fragment to validate that the
+shape of the data you're writing to the cache node is the same as the shape of
+the data required by the fragment. Under the hood, `cache.writeData`
+automatically constructs a fragment from the `data` object & `id` you pass in
+and calls `cache.writeFragment`.
+
+<h2 id="directive">@client directive</h2>
+
+Adding the `@client` directive to a field is how Apollo Link knows to resolve
+your data from the Apollo cache instead of making a network request. This
+approach is similar to other Apollo Link APIs, such as
+[`apollo-link-rest`](https://github.com/apollographql/apollo-link-rest), which
+uses the `@rest` directive to specify fields that should be fetched from a REST
+endpoint. To clarify, the `@client` and `@rest` directives never modify the
+shape of the result; rather, they specify where the data is coming from.
+
+<h3 id="combine">Combining local & remote data</h3>
+
+What's really cool about using a `@client` directive to specify client-side only
+fields is that you can actually combine local & remote data in one query. In
+this example, we're querying our user's name from our GraphQL server and their
+cart from our Apollo cache. Both the local & remote data will be merged together
+in one result.
+
+```js
+const getUser = gql`
+  query getUser($id: String) {
+    user(id: $id) {
+      id
+      name
+      cart @client {
+        product {
+          name
+          id
+        }
+      }
+    }
+  }
+`;
+```
+
+Thanks to the power of directives & Apollo Link, you'll soon be able to request
```

Comment:
What does "soon" mean here? Does it not actually work yet?

Reply:
Honestly I'm not sure haha, it depends on what's going on with `apollo-link-rest`. Server data + client-side data work together in one query though.